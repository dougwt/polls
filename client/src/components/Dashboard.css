.tabs {
  background: none;
}
.tabs .tab a,
.tabs .tab a.active,
.tabs .tab a.active:hover {
  color: #06bad6;
}
.tabs .tab a:hover {
  color: #ef476f;
}
.tabs .indicator {
  background-color: #06bad6;
}

.pagination li.active {
  background-color: #06bad6;
}
