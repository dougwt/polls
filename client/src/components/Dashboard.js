import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Section, Container, Row } from 'react-materialize';
import PollList from './polls/PollList';
import { fetchPolls } from '../actions';

const ITEMS_PER_PAGE = 15;
const MAX_PAGINATION_BUTTONS = 7;

export class Dashboard extends Component {
  componentDidMount() {
    // Ensure Materialize Tabs are initialized every time the component
    // is mounted. Without this, they only appear when the component is
    // present during the initial page load. This ensures that it works
    // properly when the user loads another page and navigates here.
    window.$('ul.tabs').tabs();

    this.props.fetchPolls();
  }

  render() {
    return (
      <Section>
        <Container>
          <Row>
            <div className="col s12">
              <h3>Dashboard</h3>
              <ul className="tabs">
                <li className="tab col s3">
                  <a href="#my">My Polls</a>
                </li>
                <li className="tab col s3">
                  <a href="#other">Other Polls</a>
                </li>
              </ul>
            </div>
          </Row>

          <Row>
            <div id="my" className="col s12">
              <PollList
                title="Polls created by you:"
                polls={(this.props.polls || []).filter(
                  poll =>
                    poll.owner &&
                    this.props.auth &&
                    poll.owner === this.props.auth._id
                )}
                itemsPerPage={ITEMS_PER_PAGE}
                maxButtons={MAX_PAGINATION_BUTTONS}
              />
            </div>
            <div id="other" className="col s12">
              <PollList
                title="Polls created by other users:"
                polls={(this.props.polls || []).filter(
                  poll =>
                    poll.owner &&
                    this.props.auth &&
                    poll.owner !== this.props.auth._id
                )}
                itemsPerPage={ITEMS_PER_PAGE}
                maxButtons={MAX_PAGINATION_BUTTONS}
              />
            </div>
          </Row>
        </Container>
      </Section>
    );
  }
}
Dashboard.propTypes = {
  auth: PropTypes.object,
  polls: PropTypes.array,
  fetchPolls: PropTypes.func
};

function mapStateToProps(state) {
  return {
    auth: state.auth,
    polls: state.poll.polls
  };
}

export default connect(mapStateToProps, { fetchPolls })(Dashboard);
