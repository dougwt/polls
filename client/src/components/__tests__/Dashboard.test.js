import React from 'react';
import { shallow } from 'enzyme';
import Dashboard from '../Dashboard';

describe('Dashboard', () => {
  const wrapper = shallow(<Dashboard />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('contains a page title', () => {
    expect(wrapper.find('h3').exists()).toBe(true);
  });

  it('contains a `My Polls` PollList', () => {
    expect(wrapper.find('#my > PollList').length).toEqual(1);
  });

  it('contains a `Other Polls` PollList', () => {
    expect(wrapper.find('#other > PollList').length).toEqual(1);
  });
});
