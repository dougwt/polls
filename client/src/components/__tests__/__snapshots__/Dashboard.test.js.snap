// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard
    fetchPolls={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Container>
        <Row>
                <div
                        className="col s12"
                >
                        <h3>
                                Dashboard
                        </h3>
                        <ul
                                className="tabs"
                        >
                                <li
                                        className="tab col s3"
                                >
                                        <a
                                                href="#my"
                                        >
                                                My Polls
                                        </a>
                                </li>
                                <li
                                        className="tab col s3"
                                >
                                        <a
                                                href="#other"
                                        >
                                                Other Polls
                                        </a>
                                </li>
                        </ul>
                </div>
        </Row>
        <Row>
                <div
                        className="col s12"
                        id="my"
                >
                        <Connect(PollList)
                                itemsPerPage={15}
                                maxButtons={7}
                                polls={Array []}
                                title="Polls created by you:"
                        />
                </div>
                <div
                        className="col s12"
                        id="other"
                >
                        <Connect(PollList)
                                itemsPerPage={15}
                                maxButtons={7}
                                polls={Array []}
                                title="Polls created by other users:"
                        />
                </div>
        </Row>
</Container>,
      "node": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Row>
            <div
                        className="col s12"
            >
                        <h3>
                                    Dashboard
                        </h3>
                        <ul
                                    className="tabs"
                        >
                                    <li
                                                className="tab col s3"
                                    >
                                                <a
                                                            href="#my"
                                                >
                                                            My Polls
                                                </a>
                                    </li>
                                    <li
                                                className="tab col s3"
                                    >
                                                <a
                                                            href="#other"
                                                >
                                                            Other Polls
                                                </a>
                                    </li>
                        </ul>
            </div>
</Row>,
          <Row>
            <div
                        className="col s12"
                        id="my"
            >
                        <Connect(PollList)
                                    itemsPerPage={15}
                                    maxButtons={7}
                                    polls={Array []}
                                    title="Polls created by you:"
                        />
            </div>
            <div
                        className="col s12"
                        id="other"
            >
                        <Connect(PollList)
                                    itemsPerPage={15}
                                    maxButtons={7}
                                    polls={Array []}
                                    title="Polls created by other users:"
                        />
            </div>
</Row>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div
              className="col s12"
>
              <h3>
                            Dashboard
              </h3>
              <ul
                            className="tabs"
              >
                            <li
                                          className="tab col s3"
                            >
                                          <a
                                                        href="#my"
                                          >
                                                        My Polls
                                          </a>
                            </li>
                            <li
                                          className="tab col s3"
                            >
                                          <a
                                                        href="#other"
                                          >
                                                        Other Polls
                                          </a>
                            </li>
              </ul>
</div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Dashboard
</h3>,
                <ul
                  className="tabs"
>
                  <li
                                    className="tab col s3"
                  >
                                    <a
                                                      href="#my"
                                    >
                                                      My Polls
                                    </a>
                  </li>
                  <li
                                    className="tab col s3"
                  >
                                    <a
                                                      href="#other"
                                    >
                                                      Other Polls
                                    </a>
                  </li>
</ul>,
              ],
              "className": "col s12",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Dashboard",
                },
                "ref": null,
                "rendered": "Dashboard",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <li
                      className="tab col s3"
>
                      <a
                                            href="#my"
                      >
                                            My Polls
                      </a>
</li>,
                    <li
                      className="tab col s3"
>
                      <a
                                            href="#other"
                      >
                                            Other Polls
                      </a>
</li>,
                  ],
                  "className": "tabs",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <a
                        href="#my"
>
                        My Polls
</a>,
                      "className": "tab col s3",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "My Polls",
                        "href": "#my",
                      },
                      "ref": null,
                      "rendered": "My Polls",
                      "type": "a",
                    },
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <a
                        href="#other"
>
                        Other Polls
</a>,
                      "className": "tab col s3",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Other Polls",
                        "href": "#other",
                      },
                      "ref": null,
                      "rendered": "Other Polls",
                      "type": "a",
                    },
                    "type": "li",
                  },
                ],
                "type": "ul",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <div
                className="col s12"
                id="my"
>
                <Connect(PollList)
                                itemsPerPage={15}
                                maxButtons={7}
                                polls={Array []}
                                title="Polls created by you:"
                />
</div>,
              <div
                className="col s12"
                id="other"
>
                <Connect(PollList)
                                itemsPerPage={15}
                                maxButtons={7}
                                polls={Array []}
                                title="Polls created by other users:"
                />
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(PollList)
                  itemsPerPage={15}
                  maxButtons={7}
                  polls={Array []}
                  title="Polls created by you:"
/>,
                "className": "col s12",
                "id": "my",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "itemsPerPage": 15,
                  "maxButtons": 7,
                  "polls": Array [],
                  "title": "Polls created by you:",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(PollList)
                  itemsPerPage={15}
                  maxButtons={7}
                  polls={Array []}
                  title="Polls created by other users:"
/>,
                "className": "col s12",
                "id": "other",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "itemsPerPage": 15,
                  "maxButtons": 7,
                  "polls": Array [],
                  "title": "Polls created by other users:",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Container>
          <Row>
                    <div
                              className="col s12"
                    >
                              <h3>
                                        Dashboard
                              </h3>
                              <ul
                                        className="tabs"
                              >
                                        <li
                                                  className="tab col s3"
                                        >
                                                  <a
                                                            href="#my"
                                                  >
                                                            My Polls
                                                  </a>
                                        </li>
                                        <li
                                                  className="tab col s3"
                                        >
                                                  <a
                                                            href="#other"
                                                  >
                                                            Other Polls
                                                  </a>
                                        </li>
                              </ul>
                    </div>
          </Row>
          <Row>
                    <div
                              className="col s12"
                              id="my"
                    >
                              <Connect(PollList)
                                        itemsPerPage={15}
                                        maxButtons={7}
                                        polls={Array []}
                                        title="Polls created by you:"
                              />
                    </div>
                    <div
                              className="col s12"
                              id="other"
                    >
                              <Connect(PollList)
                                        itemsPerPage={15}
                                        maxButtons={7}
                                        polls={Array []}
                                        title="Polls created by other users:"
                              />
                    </div>
          </Row>
</Container>,
        "node": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Row>
              <div
                            className="col s12"
              >
                            <h3>
                                          Dashboard
                            </h3>
                            <ul
                                          className="tabs"
                            >
                                          <li
                                                        className="tab col s3"
                                          >
                                                        <a
                                                                      href="#my"
                                                        >
                                                                      My Polls
                                                        </a>
                                          </li>
                                          <li
                                                        className="tab col s3"
                                          >
                                                        <a
                                                                      href="#other"
                                                        >
                                                                      Other Polls
                                                        </a>
                                          </li>
                            </ul>
              </div>
</Row>,
            <Row>
              <div
                            className="col s12"
                            id="my"
              >
                            <Connect(PollList)
                                          itemsPerPage={15}
                                          maxButtons={7}
                                          polls={Array []}
                                          title="Polls created by you:"
                            />
              </div>
              <div
                            className="col s12"
                            id="other"
              >
                            <Connect(PollList)
                                          itemsPerPage={15}
                                          maxButtons={7}
                                          polls={Array []}
                                          title="Polls created by other users:"
                            />
              </div>
</Row>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <div
                className="col s12"
>
                <h3>
                                Dashboard
                </h3>
                <ul
                                className="tabs"
                >
                                <li
                                                className="tab col s3"
                                >
                                                <a
                                                                href="#my"
                                                >
                                                                My Polls
                                                </a>
                                </li>
                                <li
                                                className="tab col s3"
                                >
                                                <a
                                                                href="#other"
                                                >
                                                                Other Polls
                                                </a>
                                </li>
                </ul>
</div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    Dashboard
</h3>,
                  <ul
                    className="tabs"
>
                    <li
                                        className="tab col s3"
                    >
                                        <a
                                                            href="#my"
                                        >
                                                            My Polls
                                        </a>
                    </li>
                    <li
                                        className="tab col s3"
                    >
                                        <a
                                                            href="#other"
                                        >
                                                            Other Polls
                                        </a>
                    </li>
</ul>,
                ],
                "className": "col s12",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Dashboard",
                  },
                  "ref": null,
                  "rendered": "Dashboard",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <li
                        className="tab col s3"
>
                        <a
                                                href="#my"
                        >
                                                My Polls
                        </a>
</li>,
                      <li
                        className="tab col s3"
>
                        <a
                                                href="#other"
                        >
                                                Other Polls
                        </a>
</li>,
                    ],
                    "className": "tabs",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <a
                          href="#my"
>
                          My Polls
</a>,
                        "className": "tab col s3",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "My Polls",
                          "href": "#my",
                        },
                        "ref": null,
                        "rendered": "My Polls",
                        "type": "a",
                      },
                      "type": "li",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <a
                          href="#other"
>
                          Other Polls
</a>,
                        "className": "tab col s3",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Other Polls",
                          "href": "#other",
                        },
                        "ref": null,
                        "rendered": "Other Polls",
                        "type": "a",
                      },
                      "type": "li",
                    },
                  ],
                  "type": "ul",
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <div
                  className="col s12"
                  id="my"
>
                  <Connect(PollList)
                                    itemsPerPage={15}
                                    maxButtons={7}
                                    polls={Array []}
                                    title="Polls created by you:"
                  />
</div>,
                <div
                  className="col s12"
                  id="other"
>
                  <Connect(PollList)
                                    itemsPerPage={15}
                                    maxButtons={7}
                                    polls={Array []}
                                    title="Polls created by other users:"
                  />
</div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(PollList)
                    itemsPerPage={15}
                    maxButtons={7}
                    polls={Array []}
                    title="Polls created by you:"
/>,
                  "className": "col s12",
                  "id": "my",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "itemsPerPage": 15,
                    "maxButtons": 7,
                    "polls": Array [],
                    "title": "Polls created by you:",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(PollList)
                    itemsPerPage={15}
                    maxButtons={7}
                    polls={Array []}
                    title="Polls created by other users:"
/>,
                  "className": "col s12",
                  "id": "other",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "itemsPerPage": 15,
                    "maxButtons": 7,
                    "polls": Array [],
                    "title": "Polls created by other users:",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
